### PR TITLE
Fixed bug: No enroll button if edx_course_key is not defined

### LINKED ISSUE #384: Bug: Don't attempt to display enroll button when edx_course_key is null
We allow courseruns to be created without an edx_course_key when there are plans to create one later. 

Unfortunately, this means we have to protect against situations when a course key is required, like providing an enroll button. 

In normal usage, this is unlikely to happen because we wouldn't pass the enrollment_start date without having a course code. But in our testing this came up.

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #384

### CODE REVIEW

**Comment by noisecapella** [comment]
Can you add a test for the change you made?

File: static/js/util/courseList.js [hunk 1]
```diff
@@ -85,8 +85,11 @@ export function makeCourseStatusDisplay(course, now = moment()) {
     let enrollmentDate = moment(firstRun.enrollment_start_date);
     if (enrollmentDate.isAfter(now, 'day')) {
       return `Enrollment starting: ${enrollmentDate.format("M/D/Y")}`;
-    } else {
+    } else if(firstRun.course_id) {
+      let courseInfoUrl = `${SETTINGS.edx_base_url}/courses/${firstRun.course_id}/about`;
```
Comment (noisecapella) [inline static/js/util/courseList.js]: It looks like this is already defined above

**Comment by annagav** [comment]
Fixed the tests!

**Comment by noisecapella** [comment]
Code looks good, testing functionality

**Comment by noisecapella** [comment]
Functionality looks good :+1: 

Can you rename the PR and commit message after you squash to be in the past tense?

**Comment by annagav** [comment]
Is this commit message better?

**Comment by noisecapella** [comment]
Can you change `btn` to `button`, and also change the PR title?

**Pull request merged**